typescript/src/api-doc-types/PropertyMemberDoc.ts

```diff
@@ -1,9 +1,38 @@
+import { Declaration, GetAccessorDeclaration, SetAccessorDeclaration, SignatureDeclaration, Symbol, SyntaxKind } from 'typescript';
+import { getDeclarationTypeText } from "../services/TsParser/getDeclarationTypeText";
+import { getParameters } from '../services/TsParser/getParameters';
 import { encodeAnchor } from '../utils/encodeAnchor';
+import { AccessorInfoDoc } from './AccessorInfoDoc';
+import { ContainerExportDoc } from './ContainerExportDoc';
 import { MemberDoc } from './MemberDoc';
 
 export class PropertyMemberDoc extends MemberDoc {
   name = this.symbol.name;
   anchor = encodeAnchor(this.name);
   id = `${this.containerDoc.id}.${this.name}`;
   aliases = this.containerDoc.aliases.map(alias => `${alias}.${this.name}` );
+  getAccessor: AccessorInfoDoc | null;
+  setAccessor: AccessorInfoDoc | null;
+
+  constructor(
+    containerDoc: ContainerExportDoc,
+    symbol: Symbol,
+    declaration: Declaration | null,
+    getAccessorDeclaration: GetAccessorDeclaration | null,
+    setAccessorDeclaration: SetAccessorDeclaration | null,
+    basePath: string,
+    namespacesToInclude: string[],
+    isStatic: boolean,
+  ) {
+    super(containerDoc, symbol, (declaration || getAccessorDeclaration || setAccessorDeclaration)!, basePath, namespacesToInclude, isStatic);
+
+    // If this property has accessors then compute the type based on that instead
+    this.getAccessor = getAccessorDeclaration && new AccessorInfoDoc('get', this, getAccessorDeclaration, basePath, namespacesToInclude);
+    this.setAccessor = setAccessorDeclaration && new AccessorInfoDoc('set', this, setAccessorDeclaration, basePath, namespacesToInclude);
+    if (this.getAccessor) {
+      this.type = this.getAccessor.type;
+    } else if (this.setAccessor) {
+      this.type = this.setAccessor.parameters[0].split(/:\s?/)[1];
```

Comment:
This doesn't seem to work properly. Right now if it's a combination of both `getter` and `setter`, then only the type of the getter will be retrieved.

But actually it should try to find the type in the `setter` if it couldn't find a type in the `getter`. Thoughts?

Reply:
What does it mean to have a getter that has no type? Oh, I guess that if you code has no explicit return type... e.g.

```typescript
class C {
  _foo: string;
  get foo() { return this._foo; }
  set foo(value: string) { this._foo = value; }
}
```

Is that the scenario? Generally in our public API docs we always give getters a return type.
In that case, then I guess we do need to look at the setter.

Reply:
I've added a commit that fixes this... @DevVersion PTAL